opentimelineio_contrib/adapters/fcpx_xml.py

```diff
@@ -162,13 +161,24 @@ def __init__(self, otio_timeline):
             self.fcpx_xml,
             "resources"
         )
-        self.event_resource = cElementTree.SubElement(
-            cElementTree.SubElement(self.fcpx_xml, "library"),
-            "event",
-            {"name": self.otio_timeline.name}
-        )
-        self.format_dictionary = {}
-        self.resource_dictionary = {}
+        if self.otio_timeline.schema_name() == "Timeline":
+            self.timelines = [self.otio_timeline]
+        else:
+            self.timelines = [
+                timeline for timeline in self.otio_timeline.each_child(
```

Comment:
I _think_ (not 100% though) you could replace this block with something like:
`self.timelines = list(self.otio_timeline.each_child(descended_from_type=otio.schema.Timeline)))`

Reply:
Yeah, that's much better! The fix has been submitted.